bee-common/bee-common/src/ord.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module that provides features related to ordering.
+
+use core::{cmp::Ordering, iter::Iterator};
+
+/// Checks if an iterator yields ordered and unique values.
+pub fn is_unique_sorted<T: Ord, I: Iterator<Item = T>>(mut iterator: I) -> bool {
```

Comment:
How does this code check for uniqueness?

Reply:
It does it by checking order AND uniqueness at the same time, two consecutive items can't be equal

Reply:
From the name, I took it to mean "checks that every item in the iterator is unique from one-another"

Reply:
That's what it does, or am I missing something ?

Reply:
But since you also check if it's ordered, you don't need a hashmap, if that was your question

Reply:
> That's what it does, or am I missing something ?

Checks all elements against all others, I mean

Reply:
Having an equal non-consecutive item would break the order so you don't need to do that